ts.js

```diff
@@ -1,14 +1,21 @@
-"use strict"
+'use strict'
 
 module.exports = {
-  parser: "@typescript-eslint/parser",
+  parser: '@typescript-eslint/parser',
   parserOptions: {
-    sourceType: "module",
+    sourceType: 'module',
+    project: './tsconfig.json' // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/parser/README.md#parseroptionsproject
   },
-  plugins: ["@typescript-eslint"],
-  extends: ["./js", "plugin:@typescript-eslint/recommended"],
+  extends: [
+    './js',
+    'standard-with-typescript'
+  ],
+  plugins: [
+    'etc'
+  ],
   rules: {
-    // Types often are recursive & no use before define is too restrctive
-    "no-use-before-define": 0
+    'no-use-before-define': 'off', // Types often are recursive & no use before define is too restrctive
+    'etc/prefer-interface': 'error', // https://ncjamieson.com/prefer-interfaces/
```

Comment:
This is cool and all, but feels like it could be a major pain when used with JSDoc syntax and when we import types across modules. Have you checked how well it works in practice ?

Maybe more gentle warning is a better option here ?

Reply:
this only applies to our types.ts files

Reply:
I also feel like this is one of the things that is silly to optimize for, TS keeps changing how they generate typedefs and it's not unreasonable to expect that they will stop emitting obsolete types and start inlining more.

Reply:
Ok never mind linked blog post mislead me into thinking that it would force defs inside the closures.

Still wondering how is this going to work with jsdoc though, because you can't even define interface in jsdoc syntax.